### DISCUSSION TITLE
How to disable WSL DNS mechanism?

### DISCUSSION THREAD

**Discussion by Tabiskabis** [microsoft/WSL#9699] [Q&A]
In windows, I use a few search domains and NRPT rules to direct the according lookups at their right server.
When I do DNS queries in WSL, they bypass the NRPT rules and any DNS query goes to every first nameserver of each network interface.
So I thought I'd use a local DNS resolver/cache as a service in windows and implement the NRPT rules as conditional forwardings instead. This works somewhat, but only if 127.0.0.1 is the only nameserver windows knows about. So basically no more DHCP for me, or WSL starts spewing and leaking queries again.
Can this be mitigated somehow? Ideally by forcing WSL to respect NRPT or make its queries otherwise behave like windows, or force it to use only my local dns service.
I also thought about using unbound as resolver on both windows and linux, so i could include a shared conffile on both. But apparently, services on WSL are not exactly stable?

Edit: Also just realised, wtf why is resolv.conf shared across wsl distributions?

**Comment by Biswa96** [comment]
Does overwriting `/etc/resolv.conf` file with your DNS IP help?

**Reply by Tabiskabis** [reply]
There is nothing to change. WSL puts the non-static IP of the virtual switch into resolv.conf, and that's the only address that makes sense to have there. Hence my title question, how to disable (or bypass) the machanism currently handles DNS.
